mlflow/pyspark/ml/__init__.py

```diff
@@ -427,6 +438,189 @@ def _log_estimator_params(param_map):
+        weakref.finalize(predict_result, clean_id, prediction_result_id)
+
+    def get_run_id_and_dataset_name_for_evaluator_call(self, pred_result_dataset):
+        """
+        Given a metric api call (include the called metric function, and call arguments)
+        Register the call information (arguments dict) into the `metric_api_call_arg_dict_list_map`
+        and return a tuple of (run_id, eval_dataset_name)
+        """
+        if id(pred_result_dataset) in self._pred_result_id_to_dataset_name_and_run_id:
+            return self._pred_result_id_to_dataset_name_and_run_id[id(pred_result_dataset)]
+        else:
+            return None, None
+
+    def gen_evaluator_info(self, evaluator):
+        class_name = _get_fully_qualified_class_name(evaluator)
+        param_map = _get_param_map(evaluator)
+        return {"evaluator_class": class_name, "params": param_map}
+
+    def register_evaluator_call(self, run_id, metric_name, dataset_name, evaluator_info):
+        evaluator_call_info_list = self._evaluator_call_info[run_id][metric_name]
+
+        index = len(evaluator_call_info_list)
+        metric_name_with_index = self.gen_name_with_index(metric_name, index)
+        metric_key = f"{metric_name_with_index}_{dataset_name}"
+
+        evaluator_call_info_list.append((metric_key, evaluator_info))
+
+        # Set the flag to true, represent the metric info in this run need update.
+        # Later when `log_eval_metric` called, it will generate a new metric_info artifact
+        # and overwrite the old artifact.
+        self._metric_info_artifact_need_update[run_id] = True
+        return metric_key
+
+    def log_post_training_metric(self, run_id, key, value):
+        """
+        Log the metric into the specified mlflow run.
+        and it will also update the metric_info artifact if needed.
+        """
+        # Note: if the case log the same metric key multiple times,
+        #  newer value will overwrite old value
+        client = mlflow.tracking.MlflowClient()
+        client.log_metric(run_id=run_id, key=key, value=value)
+        if self._metric_info_artifact_need_update[run_id]:
+            evaluator_call_list = []
+            for v in self._metric_api_call_info[run_id].values():
+                evaluator_call_list.extend(v)
+
+            evaluator_call_list.sort(key=lambda x: x[0])
+            dict_to_log = OrderedDict(evaluator_call_list)
+            client.log_dict(run_id=run_id, dictionary=dict_to_log, artifact_file="metric_info.json")
+            self._metric_info_artifact_need_update[run_id] = False
+
+
+_autologging_metrics_manager = _AutologgingMetricsManager()
+
+
+def _get_autologging_metrics_manager():
+    """
+    Get the global `_AutologgingMetricsManager` instance which holds information used in
+    post-training metric autologging. See doc of class `_AutologgingMetricsManager` for details.
+    """
+    return _autologging_metrics_manager
```

Comment:
btw what is the purpose of this function? Can we just refer to `_autologging_metrics_manager` or am I missing something?

Reply:
`variable = _get_autologging_metrics_manager` looks like we create a **new** metric manager object.

Reply:
> Can we just refer to _autologging_metrics_manager 

Yes we can.

Reply:
But I prefer keep this method `_get_autologging_metrics_manager`, it represent the pattern of "we can only get the global instance of AutologgingMetricsManager, we cannot reconstruct it or replace it.

Reply:
I think a proper way to do this is to using the `Final` type + `mypy` ([ref](https://mypy.readthedocs.io/en/stable/final_attrs.html#final-names)), but we currently don't use `mypy`. Can we just capitalize `_autologging_metrics_manager` to `_AUTOLOGGING_METRICS_MANAGER` to indicate that it's a constant and should never be changed?